src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Tizen/MediaPlayerImpl.tizen.cs

```diff
@@ -0,0 +1,438 @@
+		TaskCompletionSource<bool>? tcsForStreamInfo;
+		IVideoOutput? videoOutput;
+		Core.MediaSource? source;
+
+		public MediaPlayerImpl()
+		{
+			player = new Player();
+			player.PlaybackCompleted += OnPlaybackCompleted;
+			player.BufferingProgressChanged += OnBufferingProgressChanged;
+			player.ErrorOccurred += OnErrorOccurred;
+		}
+
+		~MediaPlayerImpl()
+		{
+			Dispose(false);
+		}
+
+		public bool UsesEmbeddingControls { get; set; }
+
+		public bool AutoPlay { get; set; }
+
+		public bool AutoStop { get; set; }
+
+		public double Volume
+		{
+			get => player.Volume;
+			set => player.Volume = (float)value;
+		}
+
+		public bool IsMuted
+		{
+			get => player.Muted;
+			set => player.Muted = value;
+		}
+
+		public bool IsLooping
+		{
+			get => player.IsLooping;
+			set => player.IsLooping = value;
+		}
+
+		public int Duration => player.StreamInfo.GetDuration();
+
+		public int Position
+		{
+			get
+			{
+				if (player.State == PlayerState.Idle || player.State == PlayerState.Preparing)
+					return 0;
+				return player.GetPlayPosition();
+			}
+		}
+
+		public DisplayAspectMode AspectMode
+		{
+			get
+			{
+				return aspectMode;
+			}
+
+			set
+			{
+				aspectMode = value;
+				ApplyAspectMode();
+			}
+		}
+
+		bool HasSource => source != null;
+
+		IVideoOutput? VideoOutput
+		{
+			get
+			{
+				return videoOutput;
+			}
+
+			set
+			{
+				if (TargetView != null)
+					TargetView.PropertyChanged -= OnTargetViewPropertyChanged;
+
+				videoOutput = value;
+
+				if (TargetView != null)
+				{
+					TargetView.PropertyChanged += OnTargetViewPropertyChanged;
+				}
+			}
+		}
+
+		VisualElement? TargetView => VideoOutput?.MediaView;
+
+		Task TaskPrepare
+		{
+			get => taskPrepare ?? Task.CompletedTask;
+			set => taskPrepare = value;
+		}
+
+		public event EventHandler? UpdateStreamInfo;
+
+		public event EventHandler? PlaybackCompleted;
+
+		public event EventHandler? PlaybackStarted;
+
+		public event EventHandler<BufferingProgressUpdatedEventArgs>? BufferingProgressUpdated;
+
+		public event EventHandler? PlaybackStopped;
+
+		public event EventHandler? PlaybackPaused;
+
+		public event EventHandler? ErrorOccurred;
+
+		public async Task<bool> Start()
+		{
+			cancelToStart = false;
+			if (!HasSource)
+				return false;
+
+			if (player.State == PlayerState.Idle)
+			{
+				await Prepare();
+			}
+
+			if (cancelToStart)
+				return false;
+
+			try
+			{
+				Device.BeginInvokeOnMainThread(() =>
+				{
+					player.Start();
+					PlaybackStarted?.Invoke(this, EventArgs.Empty);
+				});
+			}
+			catch (Exception e)
+			{
+				Log.Error($"Error On Start : {e.Message}");
+				return false;
+			}
+			return true;
+		}
+
+		public void Pause()
+		{
+			try
+			{
+				player.Pause();
+				PlaybackPaused?.Invoke(this, EventArgs.Empty);
+			}
+			catch (Exception e)
+			{
+				Log.Error($"Error on Pause : {e.Message}");
+			}
+		}
+
+		public void Stop()
+		{
+			cancelToStart = true;
+			_ = ChangeToIdleState();
```

Comment:
@brminnick will not like this. I remember that we had a conversation on how this should be implemented, Brandon can you refresh my memory here?

Reply:
Hey Jay! We need to ensure we always `await` every `Task`.

Not `await`ing a `Task` is a huge code smell, and discarding a `Task` (e.g. `_ = Task.Run(...)`) is even worse. 

This'll need to change to something like this:

```csharp
public async Task Stop()
{
	cancelToStart = true;
	await ChangeToIdleState();
	PlaybackStopped?.Invoke(this, EventArgs.Empty);
}
```

Reply:
Hey Brandon! Thanks for your review :)
Understood. I will remove discarding a `Task`. 
Thanks.